### PR TITLE
test(Theme): increase test coverage to 100%

### LINKED ISSUE #17482: Increase `Theme` unit test coverage

### PR DESCRIPTION
**Pull request by Shankar-CodeJunkie**
Closes #17482

Increase test coverage to 100% for Theme component

#### Changelog

**Changed**

* test: add tests for Global Theme function

* test: add tests for usePrefersDarkScheme function

#### Testing / Reviewing

-  Ran yarn test and see all tests are passing 
-  Ran `yarn test packages/react/src/components/Theme/__tests__/Theme-test.js --coverage --collectCoverageFrom=packages/react/src/components/Theme/index.tsx` and seen 100 % coverage
-  Increased the test coverage for the Theme component from 70 % to 100 %

```
 PASS  packages/react/src/components/Theme/__tests__/Theme-test.js
  Theme
    ✓ should render the children passed in as a prop (15 ms)
    ✓ should set the theme in context (3 ms)
  usePrefersDarkScheme
    ✓ should set see white as light (2 ms)
    ✓ should set see g10 as light (2 ms)
    ✓ should set see g90 as light (1 ms)
    ✓ should set see g100 as light (1 ms)
    ✓ should call usePrefersDarkScheme (2 ms)
  GlobalTheme
    ✓ should set the theme globally (1 ms)

-----------|---------|----------|---------|---------|-------------------
File       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------|---------|----------|---------|---------|-------------------
All files  |     100 |      100 |     100 |     100 |                   
 index.tsx |     100 |      100 |     100 |     100 |                   
-----------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       8 passed, 8 total
Snapshots:   0 total
Time:        0.947 s, estimated 1 s

```

### CODE REVIEW

File: packages/react/src/components/Theme/__tests__/Theme-test.js [hunk 1]
```diff
@@ -90,4 +95,49 @@ describe('usePrefersDarkScheme', () => {
...
+
+    expect(screen.getByTestId('default-dark')).toHaveTextContent('light');
+    expect(hooks.useMatchMedia).toHaveBeenCalledWith(
+      '(prefers-color-scheme: dark)'
+    );
+  });
+});
+
+describe('GlobalTheme', () => {
+  jest.resetModules();
```
Comment (tay1orjones) [inline packages/react/src/components/Theme/__tests__/Theme-test.js]: This could move to be within the test body instead of this outer scope

**Review by tay1orjones — COMMENTED** [review]
Looks great! Just one minor thing I noticed

File: packages/react/src/components/Theme/__tests__/Theme-test.js (hunk 1 above)
Comment (Shankar-CodeJunkie) [reply packages/react/src/components/Theme/__tests__/Theme-test.js]: Hi @tay1orjones . Yes. I have moved that line `jest.resetModules()` inside the test body. Thank you

**Review by tay1orjones — APPROVED** [review]
LGTM, thanks again!

**Review by kennylam — APPROVED** [review]

**Pull request merged**